test/test.js

```diff
@@ -325,3 +326,747 @@ test('Testing the application/x-www-form-urlencoded example case', function (t)
   t.match(snippet, /.*--data 'secret=secret\+example\+value'.*/);
   t.end();
 });
+
+const resetParameterValues = function (
+  parameter,
+  value,
+  locationOfParameter,
+  locationOfExample,
+  style,
+  explode
+) {
+  if (typeof style === 'undefined') {
+    delete parameter.style;
+  } else {
+    parameter.style = style;
+  }
+  if (typeof explode === 'undefined') {
+    delete parameter.explode;
+  } else {
+    parameter.explode = explode;
+  }
+  delete parameter.default;
+  delete parameter.example;
+  delete parameter.examples;
+
+  parameter.in = locationOfParameter;
+
+  if (locationOfExample === 'default') {
+    parameter.default = value;
+  } else if (locationOfExample === 'example') {
+    parameter.example = value;
+  } else if (locationOfExample === 'examples') {
+    parameter.examples = {
+      firstExample: {
+        summary: 'This is a summary',
+        value,
+      },
+    };
+  }
+};
+
+const runParameterTestScenario = function (
+  t,
+  api,
+  path,
+  method,
+  expectedString
+) {
+  const result = OpenAPISnippets.getEndpointSnippets(api, path, method, [
+    'shell_curl',
+  ]);
+  const snippet = result.snippets[0].content;
+  t.true(expectedString.test(snippet));
+};
+
+const runParameterTest = function (t, options) {
+  const {
+    api,
+    path,
+    method,
+    locationOfParameter,
+    parameterName,
+    value,
+    locationOfExample,
+    style,
+    explode,
+    expectedString,
+  } = options;
+
+  const apiCopy = JSON.parse(JSON.stringify(api));
+  const parameters = apiCopy.paths[path][method].parameters;
+  const parameter = parameters.find((p) => p.name === parameterName);
+  resetParameterValues(
+    parameter,
+    value,
+    locationOfParameter,
+    locationOfExample,
+    style,
+    explode
+  );
+  const result = OpenAPISnippets.getEndpointSnippets(apiCopy, path, method, [
+    'node_request',
+  ]);
+  const snippet = result.snippets[0].content;
+  t.true(
+    expectedString.test(snippet),
+    `expected '${expectedString}' in '${snippet}'`
+  );
+};
+
+// Primitive Query Parameter Tests
+
+test('Parameter in query, no style, no explode, default value 75', function (t) {
```

Comment:
cc: @musicfuel  @matttrawicki Thoughts on testing all of the combinations.

Some can be eliminated. For example `explode` only applies when the value is an array or object. But I still want to make sure I have adequate coverage.

Also, even with that in mind there are a lot of combinations: query, path and header all work slightly different. Then we have two styles `form` and `simple`, then `explode`, then the type of value primitive, array, object. Then we have the location of the example (default, example, examples).

Was thinking of creating a JSON file with all of these test scenarios. Sort of like the individual `testScenario` values I have already. But I'm not sure that' any easier to read. 

I envision a "spreadsheet" so it'd be easier to read. Any thoughts?